kube-client/src/api/portforward.rs

```diff
@@ -293,6 +315,32 @@ where
                     .await
                     .map_err(Error::SendWebSocketMessage)?;
             }
+            Message::ToPodClose(ch) => {
+                let ch = ch as usize;
+                if ch >= initialized.len() {
+                    return Err(Error::InvalidChannel(ch));
+                }
+                let port_index = ch / 2;
+                if !shutdown[port_index] {
+                    writers[port_index].shutdown().await.map_err(Error::Shutdown)?;
+                    shutdown[port_index] = true;
+
+                    closed_ports += 1;
+                }
+            }
+            Message::FromPodClose => {
+                for writer in &mut writers {
+                    writer.shutdown().await.map_err(Error::Shutdown)?;
+                }
```

Comment:
Stop the loop here after receiving WebSocket `close`?

Reply:
I was attempting to end this gracefully by waiting for the websocket stream to complete. I guess this also relates to whether we can simply stop processing the stream when we receive the close message.